src/Service.Tests/CosmosTests/QueryFilterTests.cs

```diff
@@ -262,18 +262,26 @@ public async Task TestStringMultiFiltersOnArrayTypeWithOrCondition()
             await ExecuteAndValidateResult(_graphQLQueryName, gqlQuery, dbQueryWithJoin);
         }
 
-        private async Task ExecuteAndValidateResult(string graphQLQueryName, string gqlQuery, string dbQuery)
+        private async Task ExecuteAndValidateResult(string graphQLQueryName, string gqlQuery, string dbQuery, bool ignoreBlankResults = false)
         {
             string authToken = AuthTestHelper.CreateStaticWebAppsEasyAuthToken(specificRole: AuthorizationType.Authenticated.ToString());
             JsonElement actual = await ExecuteGraphQLRequestAsync(graphQLQueryName, query: gqlQuery, authToken: authToken);
             JsonDocument expected = await ExecuteCosmosRequestAsync(dbQuery, _pageSize, null, _containerName);
-            ValidateResults(actual.GetProperty("items"), expected.RootElement);
+            ValidateResults(actual.GetProperty("items"), expected.RootElement, ignoreBlankResults);
         }
 
-        private static void ValidateResults(JsonElement actual, JsonElement expected)
+        private static void ValidateResults(JsonElement actual, JsonElement expected, bool ignoreBlankResults)
         {
             Assert.IsNotNull(expected);
             Assert.IsNotNull(actual);
+
+            if (!ignoreBlankResults)
+            {
+                // Making sure we are not asserting emply results
+                Assert.IsFalse(expected.ToString().Equals("[]"), "Expected  Response is Empty.");
```

Comment:
Do you mean to expected.ToString() should be string.empty? If yes, can't we use string.IsNullOrEmpty check on it and the actual string?

Reply:
No it means, if I am getting "empty" result from the query then, it should be considered as a valid result until unless I am mentioning it explicitly (by passing the flag) in my test.